devtools/working-with-dsproxy/working-with-dsproxy.md

```diff
@@ -0,0 +1,437 @@
+
+Seth uses an Infura RPC URL by default but you can also configure `ETH_RPC_URL` and point it to your preferred end-point.
+
+#### Create a new dapp project
+
+Create a new folder and open it
+
+```bash
+mkdir wipe-proxy && cd wipe-proxy
+```
+
+Initialize a dapp project within it
+
+```bash
+dapp init
+```
+
+Add the DSMath library to the project to use safe math operations within the script
+
+```text
+dapp install ds-math
+```
+
+#### Setup WipeProxy.sol
+
+Import the DSMath contract and add it to the WipeProxy contract
+
+```text
+import "ds-math/math.sol";
+
+contract WipeProxy is DSMath {
+
+}
+```
+
+Let's add the required interfaces to be able to interact with functions on those contracts later in the script.
+
+Add the `TubLike` interface to interact with CDPs on the Tub contract
+
+```text
+interface TubLike {
+    function wipe(bytes32, uint) external;
+    function gov() external view returns (TokenLike);
+    function sai() external view returns (TokenLike);
+    function tab(bytes32) external returns (uint);
+    function rap(bytes32) external returns (uint);
+    function pep() external view returns (PepLike);
+}
+```
+
+Add the `TokenLike` interface to interact with functions on DAI and MKR tokens
+
+```text
+interface TokenLike {
+    function allowance(address, address) external view returns (uint);
+    function balanceOf(address) external view returns (uint);
+    function approve(address, uint) external;
+    function transfer(address, uint) external returns (bool);
+    function transferFrom(address, address, uint) external returns (bool);
+}
+```
+
+Add the `PepLike` interface to read the MKRUSD price from the Pep contract
+
+```text
+interface PepLike {
+    function peek() external returns (bytes32, bool);
+}
+```
+
+Add the `UniswapExchangeLike` interface to be able to retrieve output prices of token swaps and execute them on the Uniswap exchange contracts setup for both DAI and MKR tokens
+
+```text
+interface UniswapExchangeLike {
+    function getEthToTokenOutputPrice(uint256 tokens_bought) external view returns (uint256 eth_sold);
+    function getTokenToEthOutputPrice(uint256 eth_bought) external view returns (uint256 tokens_sold);
+    function tokenToTokenSwapOutput(uint256 tokens_bought, uint256 max_tokens_sold, uint256 max_eth_sold, uint256 deadline, address token_addr) external returns (uint256  tokens_sold);
+}
+```
+
+#### Script outline
+
+Add a new function `wipeWithDai` which takes in the following inputs,
+
+* Address of the Tub contract
+* Address of the Uniswap DAI exchange contract
+* Address of the Uniswap MKR exchange contract
+* Id of the CDP in decimals. Ex: 44
+* Amount of Dai debt to pay back on the CDP
+
+```text
+function wipeWithDai(
+    address _tub,
+    address _DAIExchange,
+    address _MKRExchange,
+    uint cupid,
+    uint wad
+) 
+    public 
+{
+    // logic
+}
+```
+
+#### Checks
+
+Ensure at least some Dai debt is being wiped in the transaction using a require statement
```

Comment:
```suggestion
Within the function body, ensure at least some Dai debt is being wiped in the transaction using a require statement
```

Reply:
Important to either make this clear in the beginning with a single line that says all next subsections will be in the body of this function and/or a combination with reduced subsection title sizes (as I've shown on `Checks` and `Initialize variables`)

Reply:
Added the line to clarify that the contents of the next sections would go within the `wipeWithDai()` function